### PR TITLE
[AND-2497] AdMob adapter onAdEnd event handling

### PR DESCRIPTION
**Pull request by HuFangshuai**
This PR have been created for the ticket:https://vungle.atlassian.net/browse/AND-2497

### CODE REVIEW

File: ThirdPartyAdapters/vungle/vungle/src/main/java/com/vungle/mediation/VungleInterstitialAdapter.java [hunk 1]
```diff
@@ -137,12 +137,20 @@ public void showInterstitial() {
...
                 @Override
                 void onAdEnd(String placement, boolean wasSuccessfulView, boolean wasCallToActionClicked) {
+                    //Deprecated event
+                }
+
+                @Override
+                void onAdClick(String placement) {
+                    if (mMediationInterstitialListener != null) {
+                        mMediationInterstitialListener.onAdClicked(VungleInterstitialAdapter.this);
+                        mMediationInterstitialListener.onAdLeftApplication(VungleInterstitialAdapter.this);
```
Comment (HuFangshuai) [inline ThirdPartyAdapters/vungle/vungle/src/main/java/com/vungle/mediation/VungleInterstitialAdapter.java]: Here we invoke onAdLeftApplication() event additionally.

File: ThirdPartyAdapters/vungle/vungle/src/main/java/com/vungle/mediation/VungleInterstitialAdapter.java [hunk 2]
```diff
@@ -305,14 +313,16 @@ void onAdFailedToLoad() {
...
-                    // Only the call to action button is clickable for Vungle ads. So the
-                    // wasCallToActionClicked can be used for tracking clicks.
-                    mMediationBannerListener.onAdClicked(VungleInterstitialAdapter.this);
-                    mMediationBannerListener.onAdOpened(VungleInterstitialAdapter.this);
-                    mMediationBannerListener.onAdClosed(VungleInterstitialAdapter.this);
-                }
+                mMediationBannerListener.onAdClicked(VungleInterstitialAdapter.this);
+                mMediationBannerListener.onAdOpened(VungleInterstitialAdapter.this);
+                mMediationBannerListener.onAdClosed(VungleInterstitialAdapter.this);
+                mMediationBannerListener.onAdLeftApplication(VungleInterstitialAdapter.this);
```
Comment (HuFangshuai) [inline ThirdPartyAdapters/vungle/vungle/src/main/java/com/vungle/mediation/VungleInterstitialAdapter.java]: Here we invoke onAdLeftApplication() event additionally.

**Comment by palrahul** [comment]
target branch should be set to develop

File: ThirdPartyAdapters/vungle/vungle/src/main/java/com/google/ads/mediation/vungle/VungleMediationAdapter.java [hunk 3]
```diff
@@ -272,27 +272,53 @@ public void run() {
     public void onAdEnd(final String placementID,
                         final boolean wasSuccessfulView,
                         final boolean wasCallToActionClicked) {
+        //Deprecated event
```
[the remark this answers is not in the document]
Comment (HuFangshuai) [reply ThirdPartyAdapters/vungle/vungle/src/main/java/com/google/ads/mediation/vungle/VungleMediationAdapter.java]: This callback override from SDK, maybe remove in future.

File: ThirdPartyAdapters/vungle/vungle/src/main/java/com/vungle/mediation/VungleInterstitialAdapter.java [hunk 4]
```diff
@@ -137,16 +137,30 @@ public void showInterstitial() {
             mVungleManager.playAd(mPlacementForPlay, mAdConfig, new VungleListener() {
                 @Override
                 void onAdEnd(String placement, boolean wasSuccessfulView, boolean wasCallToActionClicked) {
+                    //Deprecated event
```
[the remark this answers is not in the document]
Comment (HuFangshuai) [reply ThirdPartyAdapters/vungle/vungle/src/main/java/com/vungle/mediation/VungleInterstitialAdapter.java]: fixed.

**Pull request merged**